lib/Doctrine/DBAL/Driver/PDOPgSql/Driver.php

```diff
@@ -53,6 +53,14 @@ public function connect(array $params, $username = null, $password = null, array
                 $pdo->setAttribute(PDO::PGSQL_ATTR_DISABLE_PREPARES, true);
             }
 
+            /* defining client_encoding via SET NAMES to avoid inconsistent DSN support
+             * - the 'client_encoding' connection param only works with postgres >= 9.1
+             * - passing client_encoding via the 'options' param breaks pgbouncer support
+             */
+            if (isset($params['charset'])) {
+              $pdo->query('SET NAMES \''.$params['charset'].'\'');
```

Comment:
@deeky666 the point being -- checking PGSQL_LIBPQ_VERSION if it's >= 9.1 to decide whether or not to use client_encoding is obviously pointless... considering it works from 8.4

@riccardonar I am afraid you're giving us bad info... you've told me 3 different versions you are running on now. Can you please give us good/full info? Can you connect to your DB server from CLI, and copy/paste the client/server versions from there?

client_encoding is working fine from an 8.4 client -> 9.3 server.

This tells me it's the SERVER version that matters (like I've been saying) -- not the client.